src/analyses/threadJoins.ml

```diff
@@ -42,7 +46,34 @@ struct
         | _ -> ctx.local (* if multiple possible thread ids are joined, none of them is must joined*)
         (* Possible improvement: Do the intersection first, things that are must joined in all possibly joined threads are must-joined *)
       )
-    | _ -> ctx.local
+    | Unknown, "__goblint_assume_join" ->
+      let id = List.hd arglist in
+      let threads = ctx.ask (Queries.EvalThread id) in
+      if TIDs.is_top threads then (
+        M.info ~category:Unsound "Unknown thread ID assume-joined, assuming ALL threads must-joined.";
+        D.bot () (* consider everything joined, D is reversed so bot is All threads *)
+      )
+      else (
+        (* elements throws if the thread set is top *)
+        let threads = TIDs.elements threads in
+        List.fold_left (fun acc tid ->
+            let joined = ctx.global tid in
+            D.union (D.add tid acc) joined
+          ) ctx.local threads
+      )
+    | _, _ -> ctx.local
```

Comment:
I think that's the case, yes, but if I remember correctly, it's also deliberate that the meaning is "assume all possibly pointed threads to definitely having been joined". There might've been discussion about it when it was merged into this PR, but it might've been necessary to make this annotation useful on some story at all since the source of that imprecision was some broader problem.
It's consistent with the top case which does that.

Reply:
Ok, I see. Should one also emit a warning for the case that `threads` is not a single unique thread?

Reply:
That's reasonable, so I added the warning.